### ISSUE TITLE
Color field in arcade shield extension doesn't work well with keyboard nav

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [microsoft/pxt-microbit#6384]
**Describe the bug**
- It doesn't select the initial value which breaks navigation
- It has two copies of black, so if you fix it to select the initial value and it's black then it selects two which is confusing.
    - From discussion the first one actually represents transparency.
- The ragged grid with an incomplete last row also breaks keyboard navigation (hover to get a starting navigation position, note you can't get to the last row)

**To Reproduce**
1. Enable keyboard nav
2. Add the arcade shield extension
3. Use a fill screen block colour field
4. Note bugs

**Expected behavior**
Navigable as in Blockly's demo which uses a similar component without these issues.

The initial selection is because getValue subverts the value representation in a way that the superclass is unaware of so when it uses it to match a position to set focus it fails to match any value.

I discussed this with @riknoll and will open a PR that implements FieldColourNumber in terms of Blockly's field-grid-dropdown which is how the most recent version of field-colour is implemented (newer than the one MakeCode uses).
